plugins/utils/spectre.nix

```diff
@@ -0,0 +1,246 @@
+        };
+        "ag" = {
+          cmd = helpers.defaultNullOpts.mkStr "ag";
+          args = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[
+					 '--vimgrep'
+					'-s'
+				  ]" ''Default arguments passed to the ag command'';
+          options =
+            option
+            ''
+              ```lua
+              {
+                      ['ignore-case'] = {
+                      	value= "-i",
+                      	icon="[I]",
+                      	desc="ignore case"
+                      },
+                      ['hidden'] = {
+                      	value="--hidden",
+                      	desc="hidden file",
+                      	icon="[H]"
+                      },
+              }
+              ```
+            '' ''You can put any ag search option you want here it can toggle with show_option function'';
+        };
+      };
+      replaceEngine = {
+        "sed" = {
+          cmd = helpers.defaultNullOpts.mkStr "sed";
+          args = helpers.defaultNullOpts.mkNullable (with types; listOf str) null ''Default arguments passed to the sed command'';
+          options =
+            option
+            ''
+              ```lua
+              {
+                      ['ignore-case'] = {
+                      	value= "--ignore-case",
+                      	icon="[I]",
+                      	desc="ignore case"
+                      },
+              }
+              ```
+            '';
+        };
+        "oxi" = {
+          cmd = helpers.defaultNullOpts.mkStr "oxi";
+          args = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]" ''Default arguments passed to the oxi command'';
+          options =
+            option
+            ''
+              ```lua
+              {
+                      ['ignore-case'] = {
+                      	value= "i",
+                      	icon="[I]",
+                      	desc="ignore case"
+                      },
+              }
+              ```
+            '';
+        };
+      };
+      default = {
+        find = {
+          cmd = helpers.defaultNullOpts.mkStr "rg" "Must be one of the options in the findEngine";
+          options = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[ignore-case]";
+        };
+        replace = {
+          cmd = helpers.defaultNullOpts.mkStr "sed" "Must be one of the options in the replaceEngine";
+          options = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]";
+        };
+      };
+      replaceVimCmd = helpers.defaultNullOpts.mkStr "cdo";
+      isOpenTargetWin = helpers.defaultNullOpts.mkBool true "Open file on opener window";
+      isInsertMode = helpers.defaultNullOpts.mkBool false "If set to true will open window in insert mode";
+      isBlockUIBreak = helpers.defaultNullOpts.mkBool false "If set to true will map backspace and enter key to avoid ui break";
+      openTemplate =
+        mkMappingsOption
```

Comment:
Also not sure what to actually use as a type here, any help would be appreciated